config/config_helpers.go

```diff
@@ -301,6 +307,93 @@ func getAWSAccountID(include *IncludeConfig, terragruntOptions *options.Terragru
 	return *identity.Account, nil
 }
 
+// Return the output from the state of another module, managed by terragrunt. This function will parse the provided
+// terragrunt config and extract the desired output from the remote state. Note that this will error if the targetted
+// module hasn't been applied yet. If you omit the output name, this will return a map of all the outputs from the
+// module.
+func getTerragruntOutput(params []string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (*cty.Value, error) {
+	// args check: make sure the expected args are passed
+	numParams := len(params)
+	if numParams == 0 {
+		return nil, errors.WithStackTrace(EmptyStringNotAllowed("the target terragrunt config parameter"))
+	}
+	if numParams > 2 {
+		return nil, errors.WithStackTrace(WrongNumberOfParams{Func: "get_output", Expected: "1 or 2", Actual: numParams})
+	}
+
+	// target config check: make sure the target config exists
+	targetConfig := params[0]
+	if util.IsDir(targetConfig) {
+		targetConfig = util.JoinPath(targetConfig, DefaultTerragruntConfigPath)
+	}
+	if !util.FileExists(targetConfig) {
+		return nil, errors.WithStackTrace(TerragruntOutputConfigNotFound{Path: targetConfig})
+	}
+
+	// use terragrunt running functions to extract the json output from the target config
+	var stdoutBuffer bytes.Buffer
+	stdoutBufferWriter := bufio.NewWriter(&stdoutBuffer)
+	targetOptions := terragruntOptions.Clone(targetConfig)
+	targetOptions.TerraformCliArgs = []string{"output", "-json"}
+	if numParams == 2 {
+		targetOptions.TerraformCliArgs = append(targetOptions.TerraformCliArgs, params[1])
+	}
+	targetOptions.Writer = stdoutBufferWriter
+	err := targetOptions.RunTerragrunt(targetOptions)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	stdoutBufferWriter.Flush()
+	jsonString := stdoutBuffer.String()
+	jsonBytes := []byte(jsonString)
+	util.Debugf(terragruntOptions.Logger, "Retrieved output from %s as json: %s", targetConfig, jsonString)
+
+	if numParams == 2 {
+		// When indexing a single output, we can return the value directly after unparsing the json.
+		retType, err := ctyjson.ImpliedType(jsonBytes)
+		if err != nil {
+			return nil, errors.WithStackTrace(TerragruntOutputParsingError{Path: targetConfig, Err: err})
+		}
+		val, err := ctyjson.Unmarshal(jsonBytes, retType)
+		if err != nil {
+			err = TerragruntOutputEncodingError{Path: targetConfig, Err: err}
+		}
+		return &val, errors.WithStackTrace(err)
+	} else {
+		// When getting all outputs, terraform returns a json with the data containing metadata about the types, so we
```

Comment:
Extract this into a separate method with unit tests

Reply:
Done